back/engines/commercial/analysis/spec/lib/auto_tagging_task_spec.rb

```diff
@@ -17,10 +17,155 @@
         state: 'succeeded',
         progress: nil
       })
-      controversial_tag = Analysis::Tag.find_by(tag_type: AutoTaggingMethod::Controversial::TAG_TYPE)
+      controversial_tag = Analysis::Tag.find_by(tag_type: Analysis::AutoTaggingMethod::Controversial::TAG_TYPE)
       expect(controversial_tag).to be_present
       expect(idea1.tags).to include(controversial_tag)
       expect(idea2.tags).not_to include(controversial_tag)
     end
   end
+
+  describe 'PlatformTopic auto_tagging' do
+    it 'works' do
+      analysis = create(:analysis)
+      att = create(:auto_tagging_task, analysis: analysis, state: 'queued', auto_tagging_method: 'platform_topic')
+      idea1 = create(:idea_with_topics, project: att.analysis.project, topics_count: 1)
+      topic1 = idea1.topics.first
+      idea2 = create(:idea_with_topics, project: att.analysis.project, topics_count: 1)
+      topic2 = idea2.topics.first
+      shared_topic = create(:topic)
+      idea1.topics << shared_topic
+      idea2.topics << shared_topic
+
+      _pre_exisiting_tag = create(:tag, tag_type: 'platform_topic', analysis: analysis, name: topic1.title_multiloc.values.first)
+
+      expect { att.execute }
+        .to change(Analysis::Tag, :count).from(1).to(3)
+
+      att.reload
+      expect(att).to have_attributes({
+        state: 'succeeded',
+        progress: nil
```

Comment:
I would have expected `1` instead of `nil` here.

Reply:
Not all tasks have progress reporting, so thought it was cleaner and less confusing to set it to `nil` for everything whenever stuff is not actually in progress. The validations make sure it's the case.